#! /usr/bin/env python

# Initialize ros and import parameter generator
PACKAGE='robotic_experiments'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# Add parameters
#       Name     Type    Level Description     	Default 	Min   	Max
gen.add("filteredForceGain",double_t,0,"Filtering weight of force torque sensor",0.9,0,1)
gen.add("velocityLimit",double_t,0,"Velocity limit [m/s]",0.3,0.3,1)
gen.add("kxy", double_t,0,"Stiffness along x-y direction",0,0,500)
gen.add("dxy", double_t,0,"Damping along x-y direction",0,0,10)
gen.add("kphi", double_t,0,"Stiffness around phi",0,0,1)
gen.add("dphi", double_t,0,"Damping around phi",0.04,0,1e-1)
gen.add("xyPositionMapping", double_t,0,"xy Position mapping",0.4,0.05,1)
gen.add("zPositionMapping", double_t,0,"z Position mapping",0.3,0.05,1)
gen.add("rollGain", double_t,0,"z Position mapping",1,0,5)
gen.add("yawGain", double_t,0,"z Position mapping",1,0,5)
gen.add("useSharedControl", bool_t, 0,"Use shared control to assist the human in holding the object",False)


exit(gen.generate(PACKAGE, "robotic_experiments", "feetTelemanipulation_params"))